x/qgb/keeper/keeper_valset.go

```diff
@@ -125,15 +123,18 @@ func (k Keeper) GetCurrentValset(ctx sdk.Context) (types.Valset, error) {
 }
 
 // normalizeValidatorPower scales rawPower with respect to totalValidatorPower
-// to take a value between 0 and 2^32 Uses BigInt operations to avoid overflow
-// errors Example: rawPower = max (2^63 - 1), totalValidatorPower = 1 validator:
-// (2^63 - 1)
+// to take a value between 0 and 2^32.
+// Uses BigInt operations to avoid overflow errors.
+// Example: rawPower = max (2^63 - 1), totalValidatorPower = 1 validator: (2^63 - 1)
 //
 //	result: (2^63 - 1) * 2^32 / (2^63 - 1) = 2^32 = 4294967296 [this is the multiplier value below, our max output]
 //
 // Example: rawPower = max (2^63 - 1), totalValidatorPower = 1000 validators with the same power: 1000*(2^63 - 1)
 //
 //	result: (2^63 - 1) * 2^32 / (1000(2^63 - 1)) = 2^32 / 1000 = 4294967
+//
+// This is using the min-max normalization from the interval [0, total validator power] to [0, 2^32].
+// Check the `PowerDiff` method under `types.validator.go` for more information.
 func normalizeValidatorPower(rawPower uint64, totalValidatorPower cosmosmath.Int) uint64 {
```

Comment:
[Question] Could you clarify the origin of the number `4294967296` which is used later in the code (in addition to what is already in the function doc)? Is the rationale behind its use for normalization explained in any specifications or ADR? I would appreciate pointers for further reading. 🙏

Reply:
This is taken from the gravity bridge implementation. 

The value comes from here: https://en.wikipedia.org/wiki/Feature_scaling

If we take this formula: 
![image](https://github.com/celestiaorg/celestia-app/assets/36426637/96e94963-9b65-4d9b-a51a-deff1045a3e0)

with: `a=0`, `b=2^32`, `min(x) = 0`, `max(x) = total validator power`, we will find the multiplier `4294967296` being  `2^32`

Reply:
Could be a good idea to link the origins of that formula in the comments

Reply:
[bf7af2d](https://github.com/celestiaorg/celestia-app/pull/1846/commits/bf7af2d18da840fbe3c3d4937279dd7a981b9001)

Reply:
Thank you for providing clarification on the formula. Just to confirm, when referring to the term "total validator power" in the equation `max(x) = total validator power,` does it represent the maximum power of the individual validators, or does it represent the combined power of all validators? According to the formula, it appears to refer to the former, but I would like to double-check for accuracy.

Reply:
> According to the formula, it appears to refer to the former,

Why is that? I think it refers to the total validator power of the whole validator set

Reply:
Nevermind, I figured the answer, it is the combined power of all the validators.

Reply:
> According to the formula, it appears to refer to the former,

> Why is that? I think it refers to the total validator power of the whole validator set

Just saw your reply, yes, that's right.

Reply:
However, based on the example in the wiki link:
". For example, suppose that we have the students' weight data, and the students' weights span [160 pounds, 200 pounds]. "
It takes the max weight of individual students as the max for the normalization, not the aggregate of all of the weights. 
This was the reason I asked whether we consider the aggregate or the max of each individual validator.  @SweeXordious